api/handlers/v1/resource.go

```diff
@@ -147,6 +147,35 @@ func (server APIServer) DeleteResource(ctx context.Context, request *entropyv1be
 	return &response, nil
 }
 
+func (server APIServer) ApplyAction(ctx context.Context, request *entropyv1beta1.ApplyActionRequest) (*entropyv1beta1.ApplyActionResponse, error) {
+	res, err := server.resourceService.GetResource(ctx, request.GetUrn())
+	if err != nil {
+		if errors.Is(err, store.ResourceNotFoundError) {
+			return nil, status.Error(codes.NotFound, "could not find resource with given urn")
+		}
+		return nil, status.Error(codes.Internal, "failed to apply fetch resource from db")
+	}
+	action := request.GetAction()
+	params := request.GetParams().GetStructValue().AsMap()
+	resultConfig, err := server.moduleService.Act(ctx, res, action, params)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to apply action to resource")
+	}
+	res.Configs = resultConfig
+	syncedResource, err := server.syncResource(ctx, res)
+	if err != nil {
+		return nil, err
+	}
+	responseResource, err := resourceToProto(syncedResource)
```

Comment:
Could this function `resourceToProto` move to `*domain.Resource` as its receiver function `ToProto`?
So when calling it we could just do `syncedResource.ToProto()`

Reply:
Wanted to keep the dependency on protobuf only till the API/handler level hence kept it here.